lib/swaps/SwapClient.ts

```
@@ -142,43 +143,72 @@ abstract class SwapClient extends EventEmitter {
+    this.setStatus(ClientStatus.ConnectionVerified);
+    this.emit('connectionVerified', {
+      newUris,
+      newIdentifier: newPubKey,
+    });
+
+  }
+
+  protected setStatus = (status: ClientStatus): void => {
     if (this.status === status) {
       return;
     }
 
     this.logger.info(`${this.constructor.name} status: ${ClientStatus[status]}`);
     this.status = status;
-    await this.setTimers();
   }
 
-  private setTimers = async () => {
-    if (this.status === ClientStatus.ConnectionVerified) {
-      if (!this.updateCapacityTimer) {
-        await this.updateCapacity();
-        this.updateCapacityTimer = setInterval(this.updateCapacity, SwapClient.CAPACITY_REFRESH_INTERVAL);
-      }
-
-      if (this.reconnectionTimer) {
-        clearTimeout(this.reconnectionTimer);
-        this.reconnectionTimer = undefined;
-      }
-    } else {
-      if (this.updateCapacityTimer) {
-        clearInterval(this.updateCapacityTimer);
-        this.updateCapacityTimer = undefined;
-      }
-      if (this.status === ClientStatus.Disconnected || this.status === ClientStatus.OutOfSync || this.status === ClientStatus.WaitingUnlock) {
-        if (!this.reconnectionTimer) {
-          this.reconnectionTimer = setTimeout(async () => {
-            await this.verifyConnection();
-            if (!this.isConnected() && this.reconnectionTimer) {
-              // if we were still not able to verify the connection, schedule another attempt
-              this.reconnectionTimer.refresh();
-            }
-          }, SwapClient.RECONNECT_TIME_LIMIT);
-        }
-      }
+  private updateCapacityTimerCallback = async () => {
+    if (this.isConnected()) {
+      await this.updateCapacity();
+    }
+  }
+
+  private reconnectionTimerCallback = async () => {
+    if (this.status === ClientStatus.Disconnected || this.status === ClientStatus.OutOfSync || this.status === ClientStatus.WaitingUnlock) {
+      try {
+        await this.verifyConnection();
+      } catch {}
```

Are we sure about silencing all errors here? Maybe `.catch(this.logger.error)` or `this.logger.trace`?

---

I did think about this some and wasn't too sure, I can put in a debug line. I think any errors are already already being logged elsewhere so I didn't want to pollute the logs.